App/Sagas/TextileSagas.ts

```diff
@@ -156,6 +154,31 @@ export function * viewThread ( action: ActionType<typeof UIActions.viewThreadReq
   yield call(NavigationService.navigate, 'ViewThread', { id: action.payload.threadId, name: action.payload.threadName })
 }
 
+export function * getUsername (contact: TextileTypes.Contact) {
+  try {
+    if (contact.username !== undefined) return
+    const uri = contact.id ? Config.TEXTILE_CAFE_URI + '/ipns/' + contact.id + '/username' : undefined
+    const response = yield call(fetch, uri)
+    const username = yield call([response, response.text])
+    yield put(ContactsActions.getUsernameSuccess(contact, username))
+  } catch (error) {
+    // nada
+  }
+}
+
+export function * addFriends ( action: ActionType<typeof UIActions.addFriendRequest> ) {
+  try {
+    const contactResult = yield call(TextileNode.getContacts)
+    const contacts = contactResult.items
+    yield put(ContactsActions.getContactsSuccess(contacts))
+    for (let contact of contacts) {
+      yield fork(getUsername, contact)
```

Comment:
Just making sure this is the intended behavior... This is going to navigate to the 'AddFriends' screen before the usernames are fetched.